src/OnboardingSPA/components/ThemeContextProvider/index.js

```diff
@@ -1,28 +1,54 @@
 import { useState, useEffect, createContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as nfdOnboardingStore } from '../../store';
+import { setFlow } from '../../utils/api/flow';
 import { THEME_DARK, THEME_LIGHT } from '../../../constants';
 
 const ThemeContext = createContext();
 
-function getPreferredColorScheme() {
+function usePreferredColorScheme() {
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+
+	if ( currentData.sitegenThemeMode ) {
+		return currentData.sitegenThemeMode;
+	}
```

Comment:
If this is '' does it fall here? Cause ig the default theme is the Device theme

Reply:
if the user toggles the theme, currentData.sitegenThemeMode will be set to that and if the user refreshes the page or revisits, the last theme will be applied.